packages/rocketchat-ui-flextab/client/tabs/userInfo.js

```diff
@@ -0,0 +1,526 @@
+			return moment(user.createdAt).format('LLL');
+		}
+	},
+
+	canDirectMessage(username) {
+		return RocketChat.authz.hasAllPermission('create-d') && (Meteor.user.username !== username);
+	},
+
+	linkedinUsername() {
+		const user = Template.instance().user.get();
+		if (user.services && user.services.linkedin && user.services.linkedin.publicProfileUrl) {
+			return s.strRight(user.services.linkedin.publicProfileUrl), '/in/';
+		}
+	},
+
+	servicesMeteor() {
+		const user = Template.instance().user.get();
+		return (user.services != null ? user.services['meteor-developer'] : undefined);
+	},
+
+	userTime() {
+		const user = Template.instance().user.get();
+		if (user.utcOffset != null) {
+			return Template.instance().now.get().utcOffset(user.utcOffset).format(RocketChat.settings.get('Message_TimeFormat'));
+		}
+	},
+
+	canRemoveUser() {
+		return RocketChat.authz.hasAllPermission('remove-user', Session.get('openedRoom'));
+	},
+
+	canMuteUser() {
+		return RocketChat.authz.hasAllPermission('mute-user', Session.get('openedRoom'));
+	},
+
+	userMuted() {
+		const room = ChatRoom.findOne(Session.get('openedRoom'));
+		const user = Template.instance().user.get();
+
+		return _.isArray(room != null ? room.muted : undefined) && (room.muted.indexOf(user != null ? user.username : undefined) !== -1);
+	},
+
+	canSetModerator() {
+		return RocketChat.authz.hasAllPermission('set-moderator', Session.get('openedRoom'));
+	},
+
+	isModerator() {
+		return !!RoomRoles.findOne({ rid: Session.get('openedRoom'), 'u._id': Template.instance().user.get()._id, roles: 'moderator' });
+	},
+
+	canSetOwner() {
+		return RocketChat.authz.hasAllPermission('set-owner', Session.get('openedRoom'));
+	},
+
+	isOwner() {
+		return !!RoomRoles.findOne({ rid: Session.get('openedRoom'), 'u._id': Template.instance().user.get()._id, roles: 'owner' });
+	},
+
+	user() {
+		return Template.instance().user.get();
+	},
+
+	hasEmails() {
+		return _.isArray(this.emails);
+	},
+
+	hasPhone() {
+		return _.isArray(this.phone);
+	},
+
+	isLoading() {
+		return Template.instance().loadingUserInfo.get();
+	},
+
+	hasAdminRole() {
+		return RocketChat.authz.hasRole(Template.instance().user.get()._id, 'admin');
+	},
+
+	active() {
+		const user = Template.instance().user.get();
+		return (user != null ? user.active : undefined);
+	},
+
+	editingUser() {
+		return Template.instance().editingUser.get();
+	},
+
+	userToEdit() {
+		const instance = Template.instance();
+		return {
+			user: instance.user.get(),
+			back(username) {
+				instance.editingUser.set();
+
+				if (username != null) {
+					const user = instance.user.get();
+					if ((user != null ? user.username : undefined) !== username) {
+						return instance.loadedUsername.set(username);
+					}
+				}
+			}
+		};
+	},
+
+	roleTags() {
+		const uid = Template.instance().user.get()._id;
+		const roles = _.union(UserRoles.findOne(uid).roles, RoomRoles.findOne({'u._id': uid, rid: Session.get('openedRoom') }).roles);
```

Comment:
Not checking if `RoomRoles.findOne({'u._id': uid, rid: Session.get('openedRoom') })` is null of undefined